root/client/src/components/table/table.js

```diff
@@ -1,29 +1,179 @@
+          'Thursday': new Date().setDate(new Date().getDate() + 2),
+          'Friday': new Date().setDate(new Date().getDate() + 3),
+          'Saturday': new Date().setDate(new Date().getDate() + 4),
+        }
+        break;
+      case 3:
+        newStateWeek = {
+          'Sunday': new Date().setDate(new Date().getDate() - 3),
+          'Monday': new Date().setDate(new Date().getDate() - 2),
+          'Tuesday': new Date().setDate(new Date().getDate() - 1),
+          'Wednesday': today[Symbol.toPrimitive]('number'),
+          'Thursday': new Date().setDate(new Date().getDate() + 1),
+          'Friday': new Date().setDate(new Date().getDate() + 2),
+          'Saturday': new Date().setDate(new Date().getDate() + 3),
+        }
+        break;
+      case 4:
+        newStateWeek = {
+          'Sunday': new Date().setDate(new Date().getDate() - 4),
+          'Monday': new Date().setDate(new Date().getDate() - 3),
+          'Tuesday': new Date().setDate(new Date().getDate() - 2),
+          'Wednesday': new Date().setDate(new Date().getDate() - 1),
+          'Thursday': today[Symbol.toPrimitive]('number'),
+          'Friday': new Date().setDate(new Date().getDate() + 1),
+          'Saturday': new Date().setDate(new Date().getDate() + 2),
+        }
+        break;
+      case 5:
+        newStateWeek = {
+          'Sunday': new Date().setDate(new Date().getDate() - 5),
+          'Monday': new Date().setDate(new Date().getDate() - 4),
+          'Tuesday': new Date().setDate(new Date().getDate() - 3),
+          'Wednesday': new Date().setDate(new Date().getDate() - 2),
+          'Thursday': new Date().setDate(new Date().getDate() -1),
+          'Friday': today[Symbol.toPrimitive]('number'),
+          'Saturday': new Date().setDate(new Date().getDate() + 1),
+        }
+        break;
+      case 6:
+        newStateWeek = {
+          'Sunday': new Date().setDate(new Date().getDate() - 6),
+          'Monday': new Date().setDate(new Date().getDate() - 5),
+          'Tuesday': new Date().setDate(new Date().getDate() - 4),
+          'Wednesday': new Date().setDate(new Date().getDate() - 3),
+          'Thursday': new Date().setDate(new Date().getDate() - 2),
+          'Friday': new Date().setDate(new Date().getDate() - 1),
+          'Saturday': today[Symbol.toPrimitive]('number'),
+        }
+        break;
+      // Need to change default
+      default: {
+        newStateWeek = {
+          'Sunday': new Date().setDate(new Date().getDate() - 1),
+          'Monday': today[Symbol.toPrimitive]('number'),
+          'Tuesday': new Date().setDate(new Date().getDate() + 1),
+          'Wednesday': new Date().setDate(new Date().getDate() + 2),
+          'Thursday': new Date().setDate(new Date().getDate() + 3),
+          'Friday': new Date().setDate(new Date().getDate() + 4),
+          'Saturday': new Date().setDate(new Date().getDate() + 5),
+        }
+        break;
+      }
+    }
```

Comment:
this doesn't need to be done in this PR (ie this is not a blocker to merging) but this can be refactored out with some sort of algorithm to better make sure of DRY code here. we can make a separate optimization/refactor ticket to handle that